test/contact_properties.js

```diff
@@ -1,164 +1,296 @@
-      let name, createdGroup
+    it('should get a property by name', function() {
+      return hubspot.contacts.properties
+        .getByName(properties.name)
+        .then(results => {
+          expect(results).to.be.an('object')
+          expect(results).to.have.a.property('name')
+        })
+    })
+  })
 
-      beforeEach(() => {
-        name = 'test_group_' + Date.now()
+  describe('delete', function() {
+    const deleteEndpoint = {
+      path: `/properties/v1/contacts/properties/named/${properties.name}`,
+      statusCode: 204,
+    }
+    fakeHubspotApi.setupServer({ deleteEndpoints: [deleteEndpoint] })
 
-        return hubspot.contacts.properties
-          .createGroup({
-            name,
-            displayName: 'Test Group',
-          })
-          .then(data => {
-            createdGroup = data
-          })
-      })
+    before(function() {
+      if (process.env.NOCK_OFF) {
+        return createTestProperty(properties)
+      }
+    })
 
-      afterEach(done => {
-        // ignore error in the case where group was deleted
-        hubspot.contacts.properties
-          .deleteGroup(name)
-          .then(() => done(), () => done())
+    it('can delete', function() {
+      return hubspot.contacts.properties.delete(properties.name).then(data => {
+        expect(data).to.be.an('undefined')
       })
+    })
+  })
 
-      it('resolves with the created group', () => {
-        expect(createdGroup).to.be.an('object')
-        expect(createdGroup).to.have.a.property('name')
-        expect(createdGroup).to.have.a.property('displayName')
-      })
+  describe('create', function() {
+    const createEndpoint = {
+      path: '/properties/v1/contacts/properties',
+      response: properties,
+    }
+    fakeHubspotApi.setupServer({ postEndpoints: [createEndpoint] })
 
-      it('can update the group with updateGroup', () => {
-        return hubspot.contacts.properties
-          .updateGroup(name, { displayName: 'Updated display name' })
-          .then(data => {
-            expect(data).to.be.an('object')
-            expect(data).to.have.a.property('name')
-            expect(data).to.have.a.property('displayName')
-          })
-      })
+    after(function() {
+      if (process.env.NOCK_OFF) {
+        return deleteTestProperty(properties.name)
+      }
+    })
 
-      it('can delete the group with deleteGroup', () => {
-        return hubspot.contacts.properties.deleteGroup(name)
+    it('should create the property', function() {
+      return hubspot.contacts.properties.create(properties).then(data => {
+        expect(data.description).to.eq(properties.description)
       })
     })
   })
 
-  describe('getByName', function() {
-    let propertyName
+  describe('update', function() {
+    const description = 'Updated display name'
+    const updateEndpoint = {
+      path: `/properties/v1/contacts/properties/named/${properties.name}`,
+      response: { ...properties, description },
```

Comment:
I'm curious about the use of the `...` syntax. Is this returning the properties hash as an array?

Reply:
It's spatting the object into this object. Checkout https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals